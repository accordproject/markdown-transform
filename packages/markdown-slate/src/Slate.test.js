/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// @ts-nocheck
/* eslint-disable no-undef */
'use strict';

const fs = require('fs');
const path = require('path');
const Value = require('slate').Value;
const CommonMark = require('@accordproject/markdown-common').CommonMark;
const SlateMark = require('./SlateMark');

let commonMark = null;
let slateMark = null;

// @ts-ignore
beforeAll(() => {
    commonMark = new CommonMark();
    slateMark = new SlateMark();
});

/**
 * Get the name and contents of all slate test files
 * @returns {*} an array of name/contents tuples
 */
function getSlateFiles() {
    const result = [];
    const files = fs.readdirSync(__dirname + '/../test/data/');

    files.forEach(function(file) {
        if(file.endsWith('.json')) {
            let contents = fs.readFileSync(__dirname + '/../test/data/' + file, 'utf8');
            result.push([file, contents]);
        }
    });

    return result;
}

describe('slate', () => {
    getSlateFiles().forEach( ([file, jsonText], index) => {
        it(`converts ${file} to concerto`, () => {
            const slateDom = JSON.parse(jsonText);
            const value = Value.fromJSON(slateDom);
            const concertoObject = slateMark.toCommonMark(value.document);
            const json = commonMark.getSerializer().toJSON(concertoObject);
            console.log('From slate', JSON.stringify(json, null, 4));

            // check no changes to the concerto
            expect(json).toMatchSnapshot();

            // load expected markdown
            const extension = path.extname(file);
            const mdFile = path.basename(file,extension);
            const expectedMarkdown = fs.readFileSync(__dirname + '/../test/data/' + mdFile + '.md', 'utf8');

            // convert the expected markdown to concerto and compare
            const expectedConcertoObject = commonMark.fromString(expectedMarkdown);
            const expectedJson = commonMark.getSerializer().toJSON(expectedConcertoObject);
            console.log('Expected JSON', JSON.stringify(expectedJson, null, 4));

            // check that ast created from slate and from the expected md is the same
            expect(json).toEqual(expectedJson);

            // now convert the expected ast back to slate and compare
            const expectedSlate = slateMark.fromCommonMark(expectedConcertoObject);
            console.log('Expected Slate', JSON.stringify(expectedSlate, null, 4));

            // check roundtrip
            expect(expectedSlate).toEqual(slateDom.document);
        });
    });
});