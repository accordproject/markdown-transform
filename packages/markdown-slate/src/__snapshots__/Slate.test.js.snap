// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slate converts bold.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Strong",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "text": "bold",
            },
          ],
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts code.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Code",
          "text": "some code",
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts codeblock.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.CodeBlock",
      "text": "this
is a
code block.
",
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "end.",
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts commonmark.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "level": "2",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "Try CommonMark",
        },
      ],
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "You can try CommonMark here.  This dingus is powered by",
        },
        Object {
          "$class": "org.accordproject.commonmark.Link",
          "destination": "https://github.com/jgm/commonmark.js",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "text": " commonmark.js",
            },
          ],
          "title": "",
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": ", the JavaScript reference implementation.",
        },
      ],
    },
    Object {
      "$class": "org.accordproject.commonmark.List",
      "delimiter": "period",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "item one",
                },
              ],
            },
          ],
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "item two",
                },
              ],
            },
          ],
        },
      ],
      "start": "1",
      "tight": "true",
      "type": "ordered",
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts headings.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "level": "2",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "H1",
        },
      ],
    },
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "level": "2",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "H2",
        },
      ],
    },
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "level": "3",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "H3",
        },
      ],
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "end.",
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts html-block.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.HtmlBlock",
      "text": "<foo>
this
is an
html block.
</foo>
end.",
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts italic.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Emph",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "text": "italic",
            },
          ],
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts link.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "a ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Link",
          "destination": "http://clause.io",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "text": "link",
            },
          ],
          "title": "",
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts ordered.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.List",
      "delimiter": "period",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "one",
                },
              ],
            },
          ],
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "two",
                },
              ],
            },
          ],
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "three",
                },
              ],
            },
          ],
        },
      ],
      "start": "1",
      "tight": "true",
      "type": "ordered",
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts paragraphs.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "one",
        },
      ],
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "two",
        },
      ],
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "three",
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts plain.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "plain",
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts quote.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.BlockQuote",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Paragraph",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "text": "this is a quote",
            },
          ],
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts text.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": "plain ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Emph",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "text": "italic",
            },
          ],
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": " ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Strong",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "text": "bold",
            },
          ],
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": " ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Emph",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Strong",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "bold-italic",
                },
              ],
            },
          ],
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "text": " ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Code",
          "text": "code",
        },
      ],
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`slate converts unordered.json to concerto 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.List",
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "one",
                },
              ],
            },
          ],
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "two",
                },
              ],
            },
          ],
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "text": "three",
                },
              ],
            },
          ],
        },
      ],
      "tight": "true",
      "type": "bullet",
    },
  ],
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;
